CONTRIBUTING.md

```diff
@@ -0,0 +1,43 @@
+# Contributing guidelines
+## Overview
+In this file you can find basic information on contributing to [nasa/fprime](https://github.com/nasa/fprime) repository. 
+## Ways of contributing
+If you want to contribute to this repository you can do it by many ways:
+- submitting PR with code improvements
+- testing software and reporting bugs
+- reporting mistakes in documentation
+- maintaining project documentation
+- reporting any issues you find in project
+- conducting code reviews to open PR
+- suggesting solutions to open issues
+
+Feel free to contribute anyway that suits your skills and you enjoy.
+
+## Where to start
+You can start by getting to know what nasa/fprime is all about:
+- going through [README](https://github.com/nasa/fprime/blob/devel/README.md) file
+- getting familiar with [documentation](https://github.com/nasa/fprime/tree/devel/docs)
+
+If you are ready to contribute your code you could start with tasks that already require help:
+- going through [open issues](https://github.com/nasa/fprime/issues)
+- helping with reviews of [open PR](https://github.com/nasa/fprime/pulls)
+
+## Branching model
+nasa/fprime is using forking branching model. If you need additional information on that model you can find it [here](https://docs.github.com/en/get-started/quickstart/fork-a-repo).
+
+### Branch naming convention
```

Comment:
When doing a fork and PR model, do we need a branch naming convention?  Is there a typical recommendation here?

Reply:
I typically do:
- feat/a-new-feature-branch
- fix/a-bugfix-branch
- hotfix/emergency-fix-branch
- release/a-release-branch

You _can_ get more granular with names like:
- docs/branch-only-changes-docs
- ci/changes-to-ci-or-tooling

But I feel like it gets more in the way than actually helping.

Reply:
@astroesteban I use a similar naming scheme too...but since these branches will live on other people's forks, do we really need to tell them how to name their branches?  The only way that branch has into F´ is through a PR, which will put it in an already established branch.

We certainly can specify a naming pattern for branches, but shoul I reject a PR with a bad branch name that is outside the repo?  I likely would not reject it, and thus I am wondering if this is needed at all.

It certainly could be the case that that we want it to help new users choose branch names even if we won't strictly enforce it. If that is the reason to include it then I am on board!

Thoughts?

Reply:
I agree with your points. I think the naming convention makes more sense when working directly on the main repo.

 Given that the pull request is the entry point to the main F' repo for contributors it doesn't really make sense to enforce a branching schema for contributors since what they do is on _their_ fork and shouldn't be dictated by the upstream F' repo since contributors "own" their fork.

Additionally, no other popular open-source project CONTRIBUTING.md I've looked at (OpenCV, TensorFlow, PyTorch, NumPy, etc) dictates a branching schema.

Reply:
I guess you might be right about branch naming it may be an overkill and should not be in contributing guidelines.
We need to remove it than. 
In general I do feel that we could really look at contributing guidelines mentioned above and I do not mind totally changing my draft as compared to atom it is not as good.
Guess the key for creating those guidelines is answering following questions:
1. what information would help newcomers easily get involved in nasa/fprime?
2. what information would impact contributors behaviour in such way that would help maintainers with project management? 

Probably too much information or information which does not bring value to contributor's or maintainer's work could be overkill and create confusion (for ex. - branch naming).